src/modules/account/RadixAccount.ts

```diff
@@ -145,55 +155,27 @@ export default class RadixAccount {
      * @returns An observable which sends 'true' whenever the account has received and processed new information form the network
      */
     public isSynced(): Observable<boolean> {
-       return this.synced.share()
-    }
-
-    public openNodeConnection = async () => {
-        this.connectionStatus.next('CONNECTING')
-
-        try {
-            this.nodeConnection = await radixUniverse.getNodeConnection(this.address.getShard())
-            this.nodeConnection.on('closed', this._onConnectionClosed)   
-
-            // Subscribe to events
-            this.atomSubscription = this.nodeConnection.subscribe(this.address.toString())
-            
-            this.atomSubscription.subscribe({
-                next: this._onAtomReceived,
-                error: error => logger.error('Subscription error:', error),
-            })
 
-            this.connectionStatus.next('CONNECTED')
-        } catch (error) {
-            logger.error(error)
+        return combineLatest(
+            radixUniverse.ledger.onSynced(this.atomObservable),
+            this.processingAtomCounter.map((value) => value === 0),
 
-            setTimeout(this._onConnectionClosed, 1000)
-        }
+            (val1, val2) => {
+                return val1 && val2
+            },
+        ).filter(isSynced => isSynced)
     }
 
-    /**
-     * Unsubscribes the node connection to the stream of past and future atoms associated with this address account
-     * 
-     * @returns A promise with the result of the unsubscription call
-     */
-    public closeNodeConnection = async () => {
-        this.connectionStatus.next('DISCONNECTED')
+    private _onAtomReceived = async (atomObservation: RadixAtomObservation) => {
+        this.processingAtomCounter.next(this.processingAtomCounter.getValue() + 1)
```

Comment:
Is this thread safe? What will this `processingAtomCounter.next(...)` provide if another thread is executed before `this.processingAtomCounter.getValue()` on this thread is called?

Reply:
There are no threads in JS, so yes, this is safe.

Reply:
Hm... but what does `async` mean then?

Reply:
Never mind, I can google this...

Reply:
It's a real fancy way to do a callback effectively, but everything always happens in the same thread